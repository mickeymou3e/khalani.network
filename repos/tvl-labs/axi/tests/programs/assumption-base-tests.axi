const a: Boolean;
const b: Boolean;

# Test duplicated assume.
assume a {
    assume a {
        !claim(a)
    }
    !right_either(b, a)
}

# Test that b doesn't hold when the assume b block is closed.
assume a {
    assume b {
        !both(a, b)
    }
    !claim(a & b)
}
