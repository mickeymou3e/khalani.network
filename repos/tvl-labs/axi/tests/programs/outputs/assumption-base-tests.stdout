Result: Nothing
Result: Nothing
Result: Proved(if(a, or(b, a)))
